src/ci/run_check.py

```diff
@@ -300,62 +297,65 @@ def runReadyToReview(moduleName, clean=False, target="agent"):
     Args:
         - moduleName: Library to be built and analyzed.
         - clean: Delete logs.
-        - target: Build type. <agent, winagent, manager>
+        - target: Build type. <agent, winagent, server>
 
     Returns:
         - None
     """
     utils.printHeader(moduleName=moduleName,
                       headerKey="rtr")
+
+    # We first run the fastest tests
     runCppCheck(moduleName=moduleName)
     runAStyleCheck(moduleName=moduleName)
+
+    # Making a full clean, downloading external dependencies and
+    # building the corresponding target with tests flag enabled
     build_tools.cleanAll()
     build_tools.makeDeps(targetName=target,
                          srcOnly=False)
     build_tools.makeTarget(targetName=target,
                            tests=True,
                            debug=True)
+
+    # Running UTs and coverage
     runTests(moduleName=moduleName)
-    build_tools.cleanFolder(moduleName=moduleName,
-                            additionalFolder="build")
-    build_tools.configureCMake(moduleName=moduleName,
-                               debugMode=True,
-                               testMode=(False, True)
-                               [moduleName != "shared_modules/utils"],
-                               withAsan=False)
+    # TODO: technical debt. Increase the coverage for winagent
+    if (target == 'winagent' and moduleName == 'data_provider') or \
+       (target == 'winagent' and moduleName == 'shared_modules/utils'):
+        utils.printInfo(msg="Skipping coverage for {} in {} target".format(
+                        moduleName, target))
+    else:
+        runCoverage(moduleName=moduleName)
+
+    # We run valgrind for all targets except Windows
+    # TODO: make it possible to run Valgrind with Wine
```

Comment:
Remove to-do comments and create issues around this.

Reply:
The issue has been created and it will be mentioned in the comment.